### PR TITLE
[centec] Support saiserver docker for centec-x86 and centec-arm64 platform

### PR DESCRIPTION
**Pull request by guxianghong**
#### Why I did it
1. Fix docker-syncd-centec-rpc.gz compile fail
2. Support Centec saiserver docker
#### How I did it
1. Modify related Makefiles in platform/centec and platform/centec-arm64

#### How to verify it
1. For centec x86 platform, use "make configure PLATFORM=centec" and "BLDENV=buster make -f Makefile.work target/docker-syncd-centec-rpc.gz" to compile
2. For centec x86 platform, use "make configure PLATFORM=centec-arm64 PLATFORM_ARCH=arm64" and "BLDENV=buster make -f Makefile.work target/docker-syncd-centec-rpc.gz" to compile
#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106

### CODE REVIEW

File: platform/centec-arm64/docker-saiserver-centec/port_config.ini [hunk 1]
```diff
@@ -0,0 +1,53 @@
+# name          lanes          alias             index     speed    fec
```
Comment (lguohan) [inline platform/centec-arm64/docker-saiserver-centec/port_config.ini:1]: do we really need this? i thought the saiserver will use the port_config.ini in the device template.

**Comment by lguohan** [comment]
@richardyu-ms to check.

File: platform/centec-arm64/docker-saiserver-centec/port_config.ini (hunk 1 above)
Comment (guxianghong) [reply platform/centec-arm64/docker-saiserver-centec/port_config.ini:1]: I'll reference to PR #8625, reads config from hwsku folder.

**Review by richardyu-ms — APPROVED** [review]

**Review by richardyu-ms — COMMENTED** [review]
no change happed in file platform/centec-arm64/tsingma-bsp/debian/rules?

**Comment by guxianghong** [comment]
> no change happed in file platform/centec-arm64/tsingma-bsp/debian/rules?
The permissions on this file will be automatically changed to 100755, so I'm submitting it along with this PR.

**Comment by guxianghong** [comment]
> no change happed in file platform/centec-arm64/tsingma-bsp/debian/rules?

The permissions on this file will be automatically changed to 100755, so I'm submitting it along with this PR.

**Comment by guxianghong** [comment]
It seems this PR is pending on "LGTM analysis: Python Expected — Waiting for status to be reported", 
Could you help to  retest it, Thanks, @richardyu-ms .

**Comment by richardyu-ms** [comment]
> It seems this PR is pending on "LGTM analysis: Python Expected — Waiting for status to be reported", Could you help to retest it, Thanks, @richardyu-ms .

looks like LGTM did not find any code for analyzing.

**Comment by richardyu-ms** [comment]
for this file `platform/centec-arm64/tsingma-bsp/debian/rules`, looks like it is not changed.

**Comment by guxianghong** [comment]
> for this file `platform/centec-arm64/tsingma-bsp/debian/rules`, looks like it is not changed.

The permission mode of this file have been changed: 100644 → 100755

**Comment by guxianghong** [comment]
> > It seems this PR is pending on "LGTM analysis: Python Expected — Waiting for status to be reported", Could you help to retest it, Thanks, @richardyu-ms .
> 
> looks like LGTM did not find any code for analyzing.

I have retest this PR by close and reopen it.  The final status is correct now.

**Comment by guxianghong** [comment]
@richardyu-ms Please help to review this PR, Thanks.

**Comment by guxianghong** [comment]
@lguohan  @richardyu-ms Please help to review and merge this PR, Thanks.

File: platform/centec/libsaithrift-dev.mk [hunk 2]
```diff
@@ -4,15 +4,16 @@ SAI_VER = 0.9.4
 
 LIBSAITHRIFT_DEV = libsaithrift-dev_$(SAI_VER)_amd64.deb
 $(LIBSAITHRIFT_DEV)_SRC_PATH = $(SRC_PATH)/sonic-sairedis/SAI
-$(LIBSAITHRIFT_DEV)_DEPENDS += $(LIBTHRIFT) $(LIBTHRIFT_DEV) $(PYTHON_THRIFT) $(THRIFT_COMPILER) $(CENTEC_SAI)
+$(LIBSAITHRIFT_DEV)_DEPENDS += $(LIBTHRIFT) $(LIBTHRIFT_DEV) $(PYTHON_THRIFT) $(THRIFT_COMPILER) $(CENTEC_SAI) $(LIBHIREDIS) $(LIBNL3_DEV) $(LIBNL_GENL3_DEV)
```
Comment (richardyu-ms) [inline platform/centec/libsaithrift-dev.mk:7]: just want to make sure, the saiserver really needs this 'LIBHIREDIS' liberary?
The saiserver just tries to expose the sai api as PRC and should be run in an environment without any other databases from upper layer.

this libsaithrift-dev.mk will be referenced by other platforms on centec, and the depends might be installed on other platforms, just want to make sure, if those liberaries, $(LIBNL3_DEV) $(LIBNL_GENL3_DEV), also needed by other centec platforms, and you verified on them? 
Or you can just added those liberaries to the related platforms in the corresponding docker-sai*.mk or docker-sync*.mk

Comment (guxianghong) [reply platform/centec/libsaithrift-dev.mk:7]: Thanks  for the feedback.  
Centec SAI needs "LIBHIREDIS", "LIBNL3_DEV" and "LIBNL_GENL3_DEV". LIBHIREDIS is used for warmboot which will store and recover SAI software data. Othre vendor may not use redis to implement SAI warmboot. "LIBNL3_DEV" and "LIBNL_GENL3_DEV" is used for sFlow.

Saiserver will link SAI library which is depend on "LIBHIREDIS", "LIBNL3_DEV" and "LIBNL_GENL3_DEV". 

I think that other centec platforms should also need these libraries according to the centec SAI implement. If we remove these libraries from libsaithrift-dev.mk, syncd-rpc may compile link fail. 
The fail log is as following:
/usr/bin/ld: /usr/lib/libsai.so: undefined reference to `genl_ctrl_resolve_grp@libnl_3'
/usr/bin/ld: /usr/lib/libsai.so: undefined reference to `nl_connect'
/usr/bin/ld: /usr/lib/libsai.so: undefined reference to `nl_socket_alloc'
/usr/bin/ld: /usr/lib/libsai.so: undefined reference to `redisConnect'
/usr/bin/ld: /usr/lib/libsai.so: undefined reference to `redisAppendCommand'
/usr/bin/ld: /usr/lib/libsai.so: undefined reference to `nl_socket_free'
/usr/bin/ld: /usr/lib/libsai.so: undefined reference to `redisCommand'
/usr/bin/ld: /usr/lib/libsai.so: undefined reference to `redisGetReply'
/usr/bin/ld: /usr/lib/libsai.so: undefined reference to `freeReplyObject'
/usr/bin/ld: /usr/lib/libsai.so: undefined reference to `redisFree'
/usr/bin/ld: /usr/lib/libsai.so: undefined reference to `redisCommandArgv'
/usr/bin/ld: /usr/lib/libsai.so: undefined reference to `nl_socket_disable_seq_check'
collect2: error: ld returned 1 exit status

**Review by richardyu-ms — APPROVED** [review]

**Comment by guxianghong** [comment]
@richardyu-ms Thanks for review, 
Please help to merge this PR @lguohan, Thanks.

**Pull request merged**